src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -512,7 +522,31 @@ void ThetaStarPathPlanner::resetAndInitializeMemberVariables(
     open_list.clear();
     closed_list.clear();
     unblocked_grid.clear();
+    line_of_sight_cache.clear();
     cell_heuristics = std::vector<std::vector<CellHeuristic>>(
         num_grid_rows,
         std::vector<CellHeuristic>(num_grid_cols, ThetaStarPathPlanner::CellHeuristic()));
 }
+
+
+unsigned long ThetaStarPathPlanner::computeMapKey(const Coordinate &coord) const
+{
+    return coord.row() + coord.col() * (num_grid_rows + 1);
```

Comment:
what's the reason for hashing the cord this way?

Reply:
A map from `unsigned long` to `bool` is around 3 times faster than a map from `Coordinate` or `std::pair<Coordinate>` to `bool`, so we can compute an unsigned long key very quickly then we get much faster lookups (2-3X)

Reply:
I'm a bit confused about why this lookup is faster. If we have a hash function for `Coordinate` which does the same work as the conversion from `Coordinate` to `long`, and then to a hash lookup, I don't see how the hash of `Coordinate` would be 2-3x slower to lookup then using a `long` directly (assuming minimal hash collisions)?